client/my-sites/plans-grid/components/features-grid/index.tsx

```diff
@@ -436,7 +438,7 @@ class FeaturesGrid extends Component< FeaturesGridType > {
 				className="plan-features-2023-grid__table-item"
 				isTableCell={ options?.isTableCell }
 			>
-				{ ! isFreePlan( planSlug ) && (
+				{ ! isFreePlan( planSlug ) && ! isFreeHostingTrial( planSlug ) && (
```

Comment:
It'd be nice if the plan objects in the `PLANS_LIST` array had some sort of `isFree` prop so we could handle this branching for _all_ free plans. Like free Jetpack too for example.